packages/analytics-types/src/diagnostic.ts

```diff
@@ -0,0 +1,9 @@
+export interface DiagnosticOptions {
+  isDisabled?: boolean;
+  serverUrl?: string;
+  apiKey?: string;
+}
+
+export interface Diagnostic extends DiagnosticOptions {
```

Comment:
I don't think we should have `Diagnostics` extend `DiagnosticsOptions`. Instead I think we can enforce the `DiagnosticsOptions` to be provided in the constructor of `BaseDiagnostic`. This will let each `Diagnostic` implementation to decide it's own necessary options.

Reply:
Remove `extends DiagnosticOptions` here. Add api key property to Diagnostic interface as it's used in core config.

Done in https://github.com/amplitude/Amplitude-TypeScript/pull/568/commits/fb672f43a99a8c7188e6c2979ed1db48bb6e6242